test/clj_kinesis_client/core_test.clj

```diff
@@ -1,24 +1,47 @@
 (ns clj-kinesis-client.core-test
   (:require [clojure.test :refer :all]
             [clj-containment-matchers.clojure-test :refer :all]
-            [clj-kinesis-client.core :refer :all]))
+            [clj-kinesis-client.core :refer :all])
+  (:import [java.util UUID]))
 
 (defonce client (create-client :endpoint "http://localhost:4567"))
+(def test-stream-name (str (UUID/randomUUID)))
+
+(defn- ensure-stream-action-fulfilled [client stream-name f]
```

Comment:
f -> assert-fn and remove the assert-fn def - you can call (assert-fn (list-streams)) from the loop